OpenRA.Renderer.Sdl2/Sdl2GraphicsDevice.cs

```diff
@@ -308,6 +308,27 @@ public void SetLineWidth(float width)
 			ErrorHandler.CheckGlError();
 		}
 
+		public Bitmap TakeScreenshot()
+		{
+			var rect = new Rectangle(Point.Empty, size);
+			var bitmap = new Bitmap(rect.Width, rect.Height, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+			var data = bitmap.LockBits(new Rectangle(Point.Empty, rect.Size),
+				System.Drawing.Imaging.ImageLockMode.WriteOnly, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+
+			try
+			{
+				GL.ReadPixels(rect.X, rect.Y, rect.Width, rect.Height, PixelFormat.Bgra, PixelType.UnsignedByte, data.Scan0);
+				GL.Finish();
+			}
+			finally
+			{
+				bitmap.UnlockBits(data);
+			}
+
+			bitmap.RotateFlip(RotateFlipType.RotateNoneFlipY);
```

Comment:
I would comment this line. It's confusing if you're not already aware bitmaps are upside-down by default.